packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -213,7 +223,9 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
         if (!isExternal.value) {
           const routerLinkProps: Record<string, any> = {
             ref: process.server ? undefined : (ref: any) => { el!.value = ref?.$el },
-            to: to.value,
+            to: options?.trailingSlashBehavior === 'append' || options.trailingSlashBehavior === 'remove'
+              ? resolveTrailingSlashBehavior(router.resolve(to.value))
+              : to.value,
```

Comment:
idea: Could we reduce that down to a truth check?
```suggestion
            to: options?.trailingSlashBehavior
              ? resolveTrailingSlashBehavior(router.resolve(to.value))
              : to.value,
```

Reply:
In this case we'll have to send `router` as the second argument to the resolver function, add typings for `router` etc. To me it looks more complicated, so I decided to keep this check here and just send a resolved `Route` object.
Also current version helps to quickly recognize that `to` is not mutated if trailing slash strategy is not defined.

Reply:
To clarify what I mean: if we send resolved `route` and in the resolver based on no strategy defined we should just return original `to` value - we cannot do that.